src/mlpack/bindings/go/print_doc_functions_impl.hpp

```diff
@@ -0,0 +1,544 @@
+{
+  // Nothing to do.
+}
+
+/**
+ * Assemble a vector of string tuples indicating parameter names and what should
+ * be printed for them.  (For output parameters, we just need to print the
+ * value.)
+ */
+template<typename T, typename... Args>
+void GetOptions(
+    std::vector<std::tuple<std::string, std::string>>& results,
+    const std::string& paramName,
+    const T& value,
+    Args... args)
+{
+  // Determine whether or not the value is required.
+  if (CLI::Parameters().count(paramName) > 0)
+  {
+    std::ostringstream oss;
+    oss << value;
+    results.push_back(std::make_tuple(paramName, oss.str()));
+    GetOptions(results, args...);
+  }
+  else
+  {
+    // Unknown parameter!
+    throw std::runtime_error("Unknown parameter '" + paramName + "' " +
+        "encountered while assembling documentation!  Check PROGRAM_INFO() " +
+        "declaration.");
+  }
+}
+
+// Recursion base case.
+inline std::string PrintOutputOptions() { return ""; }
+
+template<typename... Args>
+std::string PrintOutputOptions(Args... args)
+{
+  // Get the list of output options for the binding.
+  std::vector<std::string> outputOptions;
+  for (auto it = CLI::Parameters().begin(); it != CLI::Parameters().end(); ++it)
+  {
+    const util::ParamData& d = it->second;
+    if (!d.input)
+      outputOptions.push_back(it->first);
+  }
+
+  // Now get the full list of output options that we have.
+  std::vector<std::tuple<std::string, std::string>> passedOptions;
+  GetOptions(passedOptions, args...);
+
+  // Next, iterate over all the options.
+  std::ostringstream oss;
+  for (size_t i = 0; i < outputOptions.size(); ++i)
+  {
+    // Does this option exist?
+    bool found = false;
+    size_t index = passedOptions.size();
+    for (size_t j = 0; j < passedOptions.size(); ++j)
+    {
+      if (outputOptions[i] == std::get<0>(passedOptions[j]))
+      {
+        found = true;
+        index = j;
+        break;
+      }
+    }
+
+    if (found)
+    {
+      // We have received this option, so print it.
+      if (i == 0)
+      {
+        oss << std::get<1>(passedOptions[index]);
+      }
+      else if (i > 0)
+      {
+        oss << ", ";
+        oss << std::get<1>(passedOptions[index]);
+      }
+    }
+    else
+    {
+      // We don't care about this option.
+      if (i == 0)
+      {
+        oss << "_";
+      }
+      else if (i > 0)
+      {
+        oss << ", _";
+      }
+    }
+  }
+
+  return oss.str();
+}
+
+/**
+ * Given a name of a binding and a variable number of arguments (and their
+ * contents), print the corresponding function call.
+ */
+template<typename... Args>
+std::string ProgramCall(const std::string& programName, Args... args)
+{
+  std::string result = "";
+  std::string goProgramName = CamelCase(programName, false);
+
+  // Initialize the method parameter structure
+  std::ostringstream oss;
+  oss << "    // Initialize optional parameters for " << goProgramName << "()."
```

Comment:
```suggestion
  oss << "// Initialize optional parameters for " << goProgramName << "()."
```

I don't think that there should be any need for indents here.